src/mctp-bench.c

```diff
@@ -0,0 +1,413 @@
+#define _XOPEN_SOURCE 700
+#include <bits/time.h>
+#include <err.h>
+#include <errno.h>
+#include <poll.h>
+#include <signal.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/ioctl.h>
+#include <sys/poll.h>
+#include <sys/socket.h>
+#include <time.h>
+#include <unistd.h>
+
+#include "mctp.h"
+
+struct mctp_bench_send_args {
+        mctp_eid_t eid;
+        size_t len;
+        int net;
+};
+
+struct msg_header {
+        uint16_t magic;
+        uint32_t seq_no;
+};
+
+struct mctp_stats {
+        size_t total_received_len, curr_packet_len;
+        uint32_t prev_seq_no;
+        double elapsed_time;
+        unsigned long msgs_dropped, msg_count, invalid_payloads;
+};
+
+struct recv_ctx {
+        struct mctp_stats stats;
+        struct timespec start_time, current_time;
+        unsigned char *buf;
+        volatile bool started_recv_flag;
```

Comment:
Don't need `volatile` (nothing is modifying memory behind the compiler's back, which is the case where `volatile` is _sometimes_ right to use)